contracts/zero-ex/test/features/meta_transactions_test.ts

```diff
@@ -0,0 +1,518 @@
+                    mtx.maxGasPrice,
+                ),
+            );
+        });
+
+        it('fails if expired', async () => {
+            const mtx = getRandomMetaTransaction({
+                expirationTime: new BigNumber(Math.floor(_.now() / 1000 - 60)),
+            });
+            const mtxHash = getExchangeProxyMetaTransactionHash(mtx);
+            const signature = await signMetaTransactionAsync(mtx);
+            const callOpts = {
+                gasPrice: mtx.maxGasPrice,
+                value: mtx.value,
+            };
+            const tx = feature.executeMetaTransaction(mtx, signature).awaitTransactionSuccessAsync(callOpts);
+            return expect(tx).to.revertWith(
+                new ZeroExRevertErrors.MetaTransactions.MetaTransactionExpiredError(
+                    mtxHash,
+                    undefined,
+                    mtx.expirationTime,
+                ),
+            );
+        });
+
+        it('fails if wrong sender', async () => {
+            const requiredSender = randomAddress();
+            const mtx = getRandomMetaTransaction({
+                sender: requiredSender,
+            });
+            const mtxHash = getExchangeProxyMetaTransactionHash(mtx);
+            const signature = await signMetaTransactionAsync(mtx);
+            const callOpts = {
+                gasPrice: mtx.maxGasPrice,
+                value: mtx.value,
+            };
+            const tx = feature.executeMetaTransaction(mtx, signature).awaitTransactionSuccessAsync(callOpts);
+            return expect(tx).to.revertWith(
+                new ZeroExRevertErrors.MetaTransactions.MetaTransactionWrongSenderError(
+                    mtxHash,
+                    sender,
+                    requiredSender,
+                ),
+            );
+        });
+
+        it('fails if signature is wrong', async () => {
+            const mtx = getRandomMetaTransaction({ signer: signers[0] });
+            const mtxHash = getExchangeProxyMetaTransactionHash(mtx);
+            const signature = await signMetaTransactionAsync(mtx, signers[1]);
+            const callOpts = {
+                gasPrice: mtx.maxGasPrice,
+                value: mtx.value,
+            };
+            const tx = feature.executeMetaTransaction(mtx, signature).awaitTransactionSuccessAsync(callOpts);
+            return expect(tx).to.revertWith(
+                new ZeroExRevertErrors.MetaTransactions.MetaTransactionInvalidSignatureError(
+                    mtxHash,
+                    signature,
+                    new ZeroExRevertErrors.SignatureValidator.SignatureValidationError(
+                        ZeroExRevertErrors.SignatureValidator.SignatureValidationErrorCodes.WrongSigner,
+                        mtxHash,
+                        signers[0],
+                        signature,
+                    ).encode(),
+                ),
+            );
+        });
+    });
+
+    describe('executeMetaTransactions()', () => {
```

Comment:
maybe add a check that you can't execute the same mtx twice via `executeMetaTransactions`